### ISSUE TITLE
Access to Plataform.sh behind a proxy

### ISSUE THREAD

**Issue by rpayanm** [platformsh/platformsh-cli#294]
I installed platformsh-cli whitout problems, but when I try connect to Plataform.sh I get this error:

```
[~]$ platform
Welcome to Platform.sh!

Please log in using your Platform.sh account

Your email address: hello@example.com
Your password:

[GuzzleHttp\Exception\ConnectException]                                      
  Error creating resource: [message] fopen(https://marketplace.commerceguys.c  
  om/oauth2/token): failed to open stream: Connection refused                  
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 405                                                                   
  [message] Undefined variable: http_response_header                           
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 406

[GuzzleHttp\Exception\ConnectException]                                      
  Error creating resource: [message] fopen(https://marketplace.commerceguys.c  
  om/oauth2/token): failed to open stream: Connection refused                  
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 405                                                                   
  [message] Undefined variable: http_response_header                           
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 406

[GuzzleHttp\Ring\Exception\RingException]                                    
  Error creating resource: [message] fopen(https://marketplace.commerceguys.c  
  om/oauth2/token): failed to open stream: Connection refused                  
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 405                                                                   
  [message] Undefined variable: http_response_header                           
  [file] /home/rpayanm/.composer/vendor/guzzlehttp/ringphp/src/Client/StreamH  
  andler.php                                                                   
  [line] 406 

```

I am behind a proxy and I config the terminal for work with my proxy:

```
[~]$ echo $http_proxy
http://192.168.1.1:3128
[~]$ echo $https_proxy
http://192.168.1.1:3128
[~]$ echo $ftp_proxy
http://192.168.1.1:3128
```

How I can fix that?

**Comment by pjcdawkins** [comment]
According to the [Guzzle documentation](http://guzzle.readthedocs.org/en/latest/clients.html#environment-variables), it may work to define the environment variable `HTTPS_PROXY`. So that would just be an uppercase version of your current variable.

Try
```bash
export HTTPS_PROXY=$https_proxy
platform
```

**Comment by rpayanm** [comment]
I get a same error :(
My variables:
```
[~]$ echo $http_proxy
http://192.168.1.1:3128
[~]$ echo $https_proxy
http://192.168.1.1:3128
[~]$ echo $ftp_proxy
http://192.168.1.1:3128
[~]$ echo $HTTP_PROXY
http://192.168.1.1:3128
[~]$ echo $HTTPS_PROXY
http://192.168.1.1:3128
[~]$ echo $FTP_PROXY
http://192.168.1.1:3128
```

**Comment by pjcdawkins** [comment]
That really does seem to be the way to get Guzzle to work behind a proxy, so I wonder if that's not the issue.

Try unsetting the proxy just for one command:
```bash
http_proxy= https_proxy= HTTP_PROXY= HTTPS_PROXY= platform
```

**Comment by rpayanm** [comment]
I can't connect yet.

With my proxy I can download symfony2 with this patch, maybe this can help:

https://github.com/symfony/symfony-installer/issues/124
https://github.com/symfony/symfony-installer/pull/132/files

**Comment by pjcdawkins** [comment]
I have implemented something in #298 - I'm not sure how to test it, but you could try it yourself with:

```
git clone --branch proxy-support https://github.com/pjcdawkins/platformsh-cli.git cli-proxy-support
cd cli-proxy-support
composer install --no-dev
./platform
```

It uses the HTTPS_PROXY environment variable (none of the others).

**Comment by rpayanm** [comment]
I set my proxy as you said:
```
export $HTTPS_PROXY=http://192.168.1.1:3128/
```
Then throw a error:
```
[GuzzleHttp\Exception\RequestException]                                      
  Error creating resource: [message] fopen(https://marketplace.commerceguys.c  
  om/oauth2/token): failed to open stream: Unable to find the socket transpor  
  t "socks" - did you forget to enable it when you configured PHP?             
  [file] /home/rpayanm/cli-proxy-support/vendor/guzzlehttp/ringphp/src/Client  
  /StreamHandler.php                                                           
  [line] 404                                                                   
  [message] Undefined variable: http_response_header                           
  [file] /home/rpayanm/cli-proxy-support/vendor/guzzlehttp/ringphp/src/Client  
  /StreamHandler.php                                                           
  [line] 405                                                                   

  [GuzzleHttp\Ring\Exception\RingException]                                    
  Error creating resource: [message] fopen(https://marketplace.commerceguys.c  
  om/oauth2/token): failed to open stream: Unable to find the socket transpor  
  t "socks" - did you forget to enable it when you configured PHP?             
  [file] /home/rpayanm/cli-proxy-support/vendor/guzzlehttp/ringphp/src/Client  
  /StreamHandler.php                                                           
  [line] 404                                                                   
  [message] Undefined variable: http_response_header                           
  [file] /home/rpayanm/cli-proxy-support/vendor/guzzlehttp/ringphp/src/Client  
  /StreamHandler.php                                                           
  [line] 405
```
I google it the error and I find the solution:
instead of
```
export $HTTPS_PROXY=http://192.168.1.1:3128/
```
must be:
```
export $HTTPS_PROXY=192.168.1.1:3128/
```
Then everything works! But... I think that **http://** is like a standard and this var(HTTPS_PROXY) is used by other programs, then you can please fix it for this works:
```
export $HTTPS_PROXY=http://192.168.1.1:3128/
```
instead of:
```
export $HTTPS_PROXY=192.168.1.1:3128/
```
Thank you.

**Comment by pjcdawkins** [comment]
Shouldn't the HTTPS_PROXY begin with `https://`? or `ssl://`?

**Comment by pjcdawkins** [comment]
OK, I've been advised that http:// is valid for HTTPS proxies... I've updated the PR #298 with my guesses, I think Guzzle may need me to translate `http://` to `tcp://`

**Comment by rpayanm** [comment]
How I test the PR #298 ?

**Comment by pjcdawkins** [comment]
The same thing as before, but clone it into a new directory perhaps, as it has been rebased.
```
git clone --branch proxy-support https://github.com/pjcdawkins/platformsh-cli.git cli-proxy-support-2
cd cli-proxy-support-2
composer install --no-dev
./platform
```

**Comment by rpayanm** [comment]
wow!!! nice work like a charm!

thank you.

**Comment by pjcdawkins** [comment]
Released in version 2.0.7 - update with:

```bash
composer global update
```
